tools/tools_shared_functions.py

```diff
@@ -358,6 +359,23 @@ def cross_walk_gval_fim(metric_df: pd.DataFrame, cell_area: int, masked_count: i
     return {x: y for x, y in zip(metric_df.columns, metric_df.values[0])}
 
 
+def delete_and_garbage_collect(objects_to_delete: list):
+    """
+    Deletes and garbage collects objects to free memory
+
+    Prammeters
+    ----------
+    object_to_delete : list
+        List of objects to delete and garbage collect
+
+    """
+
+    for obj in objects_to_delete:
+        del obj
```

Comment:
Note that this only deletes the reference to the object, not the object itself.

Reply:
The attempt to manually manage memory here I think is misguided. Python is already doing most of this for you automatically. There is often no need to manually invoke expensive gc collections, though there are some exceptions.

Reply:
ya.. There are a few places it might help though (I think).  We have some areas where we keep some very large and multiple datasets and lists open simultaneously. There are lots of places, a lot in other FIM Files and not so much here, we can drop objects after we need them, or pull out what we need to a subset object so we can let the bigger one be cleaned up earlier. Most of that is on a file by file, line by  line basis. The addition of so many large dataset open at the same time are creating pretty big problems. (Granted.. this statement mostly applies to other FIM code files, and not so much this PR files. But bringing awareness to it helps. :) (I think).  Code in multi processing have shown to be a bit slower on garbage collections and we have seen a few impacts on it.

Reply:
timing on the firing of the python garbage collection is async and minor delays can be seen on rare occasions.